src/bwdist.jl

```diff
@@ -26,20 +30,45 @@ See also: [`distance_transform`](@ref).
 Transforms of Binary Images in Arbitrary Dimensions' [Maurer et al.,
 2003] (DOI: 10.1109/TPAMI.2003.1177156)
 """
-function feature_transform(I::AbstractArray{Bool,N}, w::Union{Nothing,NTuple{N}}=nothing) where N
+function feature_transform(img::AbstractArray{<:Union{Bool,AbstractGray{Bool}},N};
+                           weights::Union{Nothing,NTuple{N}}=nothing,
+                           nthreads::Int = length(img) < 1000 ? 1 : Threads.nthreads()) where N
```

Comment:
Curious to ask, does this mean that ComputationalResources.jl is considered deprecated?

Reply:
I'm glad you ask. I'm not sure how to think about that. It was added before keyword arguments to give meaning to options that otherwise might be too simple to be unambiguous (as you've noted, `imfilter` has a complicated dispatch hierarchy and could be cleaned up with keyword arguments). That said, it does provide facilities for passing options, and that's not a bad thing. I could incorporate it here. Just not quite sure what to think...

Reply:
From a more general perspective, I could imagine that algorithm `f` is expected to accept `img::CuArray` and calls the CUDA implementation, in which there won't be `nthreads` keyword.

The difficulty in ComputationalResources is that if we don't separate the interface dispatch from the implementation dispatch, we need to handle the complicated dispatch hierarchy for every function. I personally think it is affordable to maintain one or two complicated dispatch hierarchies using ComputationalResources. So for example, in ImageBinarization we can only configure CR for `binarize`/`binarize!`, and leave all the implementation dispatch to concrete algorithms, e.g., https://github.com/zygmuntszpak/ImageBinarization.jl/blob/0c0c5eaefa997f07746c24887c9ed5fad6199d9b/src/algorithms/niblack.jl#L84-L85

Said this, the current `nthreads` keyword version looks good to me right now. We can reraise this discussion when we have more concrete issues.